servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/IterableMergeCompletable.java

```diff
@@ -37,49 +35,37 @@
 
     @Override
     public MergeSubscriber apply(final Subscriber subscriber) {
-        if (others instanceof Collection) {
-            return new FixedCountMergeSubscriber(subscriber, 1 + ((Collection) others).size(), delayError);
-        } else {
-            return new DynamicCountSubscriber(subscriber, delayError);
-        }
+        return new DynamicCountSubscriber(subscriber, delayError);
     }
 
     @Override
     void doMerge(final MergeSubscriber subscriber) {
-        if (subscriber instanceof DynamicCountSubscriber) {
-            int count = 1;
-            for (Completable itr : others) {
-                ++count;
-                itr.subscribeInternal(subscriber);
-            }
-            ((DynamicCountSubscriber) subscriber).setExpectedCount(count);
-        } else {
-            for (Completable itr : others) {
-                itr.subscribeInternal(subscriber);
-            }
+        long count = 1;
+        for (Completable itr : others) {
+            ++count;
+            itr.subscribeInternal(subscriber);
         }
+        ((DynamicCountSubscriber) subscriber).setExpectedCount(count);
```

Comment:
If you prefer to keep only `DynamicCountSubscriber`, consider adding a generic type for `AbstractMergeCompletableOperator` to avoid cast here.

Reply:
I experimented with this but you can't have a generic sub-type with an inner class. I'll move the inner classes so I can avoid the cast.